src/report-table/report-table.component.tsx

```diff
@@ -172,4 +175,27 @@ const ReportTable = props => {
   )
 }
 
+const mergeMultipleTestsBasedOnUploadedFile = (
+  diagnosticReports: Array<ReportEntry>,
+): Array<ReportEntry> => {
+  const documentUrls: Array<string> = []
+  const uniqueDiagnosticReports: Array<ReportEntry> = []
+  for (let index = 0; index < diagnosticReports?.length; index++) {
+    const diagnosticReport: ReportEntry = diagnosticReports[index]
+    const documentUrl: string = diagnosticReport.resource.presentedForm[0].url
+    if (documentUrls.includes(documentUrl)) {
+      const existingReports: Array<ReportEntry> = uniqueDiagnosticReports.filter(
+        report => report.resource.presentedForm[0].url === documentUrl,
+      )
+      if (existingReports.length === 1) {
+        existingReports[0].resource.code.coding[0].display = `${existingReports[0].resource.code.coding[0].display}, ${diagnosticReport.resource.code.coding[0].display}`
+      }
+    } else {
+      documentUrls.push(documentUrl)
+      uniqueDiagnosticReports.push(diagnosticReport)
+    }
+  }
+  return uniqueDiagnosticReports
+}
```

Comment:
While the code is imperative, it still needed eyeballing up and down to get the gist of what we are trying to do here... you can refactor it to be more functional with map-reduce

1) Find unique report references - dont merge this with the rest of deduplication code
2) Your unique report references (url) would be the guide towards deduplication (map). For every unique report reference (url) accumulate the code comma seperated. 
3) lastly you are deduplicating - this name sounds more like what function is litrally doing  `mergeMultipleTestsBasedOnUploadedFile`

```
function url(resource: Resource) {
  return resource.presentedForm[0].url
}
function code(resource: Resource) {
  return resource.code.coding[0].display
}

function getUniqueReportUrls(diagnosticReport: ReportEntry[]) {
  return new Set<string>(diagnosticReport.map(report => url(report.resource)))
}

function dedupe(diagnosticReport: Array<ReportEntry>) {
  return Array.from(getUniqueReportUrls(diagnosticReport)).map(reportUrl =>
    diagnosticReport.reduce<ReportEntry | undefined>((acc, curr) => {
      if (url(curr.resource) === reportUrl) {
        if (acc)
          acc.resource.code.coding[0].display = `${code(acc.resource)}, ${code(
            curr.resource,
          )}`
        else return curr
      }
      return acc
    }, undefined),
  )
}
```

Reply:
changes have been made